lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -252,41 +259,42 @@ def __get_k8s_mountable_volumes(self, job_wrapper):
         }
         return [k8s_mountable_volume]
 
-    def __get_k8s_containers(self, job_wrapper):
+    def __get_k8s_containers(self, ajs):
         """Fills in all required for setting up the docker containers to be used, including setting a pull policy if
            this has been set.
         """
         k8s_container = {
-            "name": self.__get_k8s_container_name(job_wrapper),
-            "image": self._find_container(job_wrapper).container_id,
+            "name": self.__get_k8s_container_name(ajs.job_wrapper),
+            "image": self._find_container(ajs.job_wrapper).container_id,
             # this form of command overrides the entrypoint and allows multi command
             # command line execution, separated by ;, which is what Galaxy does
             # to assemble the command.
-            # TODO possibly shell needs to be set by job_wrapper
-            "command": ["/bin/bash", "-c", job_wrapper.runner_command_line],
-            "workingDir": job_wrapper.working_directory,
+            "command": [ajs.job_wrapper.shell],
+            "args": ["-c", ajs.job_file],
+            "workingDir": ajs.job_wrapper.working_directory,
             "volumeMounts": [{
                 "mountPath": self.runner_params['k8s_persistent_volume_claim_mount_path'],
                 "name": self._galaxy_vol_name
             }]
         }
 
-        resources = self.__get_resources(job_wrapper)
+        resources = self.__get_resources(ajs.job_wrapper)
         if resources:
+            envs = []
+            if 'limits' in resources:
+                limits = resources['limits']
+                if 'memory' in limits:
+                    envs.append({'name': 'GALAXY_MEMORY_MB', 'value': str(ByteSize(limits['memory']).to_unit('M', as_string=False))})
```

Comment:
limits normally come provided as Gi normally, are you sure that this part will work fine? Or ByteSize finds out the unit in which the require/limits are given?